django/core/cache/backends/redis.py

```diff
@@ -218,15 +218,17 @@ def set_many(self, data, timeout=DEFAULT_TIMEOUT, version=None):
         for key, value in data.items():
             key = self.make_and_validate_key(key, version=version)
             safe_data[key] = value
-        self._cache.set_many(safe_data, self.get_backend_timeout(timeout))
+        if safe_data:
+            self._cache.set_many(safe_data, self.get_backend_timeout(timeout))
```

Comment:
Is there a reason to do this with `safe_data` instead of passed `data`? (the same in `delete_many()`), e.g.
```python
def set_many(self, data, timeout=DEFAULT_TIMEOUT, version=None):
    if not data:
        return []
    ...
```

Reply:
Not really no.. Now that i think of it this will also address issues where data or keys (delete_many) don't have the expected type.. e.g. keys been None or data been None..

Is it ok to modify the pr as per you suggestion and adjust the 2 tests accordingly to cover for keys or data been None?

Reply:
> Not really no.. Now that i think of it this will also address issues where data or keys (delete_many) don't have the expected type.. e.g. keys been None or data been None..

Why? `data`/`keys` are not falsey when they contains `None` :thinking:

Reply:
I mean introducing your suggestions to place the check at the top of the method will also resolve issues where e.g data or keys parameter passed by the user in the respective methods is None.

In that case the relevant code block e.g

for key, value in data.items ....

Will fail. This is why it's a better approach.


I'll proceed with the changes.

Thanks for the suggestion

Reply:
Ahh sorry, I misunderstood. Thanks :+1: